### PR TITLE
js-oop

### PR DESCRIPTION
**Pull request by merowing**
### js-oop

[Demo](https://merowing.github.io/js-oop/) |
[Code base](https://github.com/merowing/js-oop)

The code is submitted in a dedicated feature branch.

Only code files are submitted.

Please, review.

### CODE REVIEW

**Comment by merowing** [comment]
Review and fix the code. All done.

File: submissions/merowing/js-oop/app.js [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+        this.speed = randomNumber();
+    }
+
+    if(
+        this.x + 51 > player.x && // enemy in front of player
+        this.x - player.x <= 51 && // enemy behind of player
+        this.y - player.y === 0 // enemy and player on the same line
+    ) {
+        player.x = defaultPlayerLocation.x;
+        player.y = defaultPlayerLocation.y;
```
Comment (OleksiyRudenko) [inline submissions/merowing/js-oop/app.js:25]: Item 4.iii of the reuirements.

File: submissions/merowing/js-oop/app.js [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+    if(keyCode === 'down' && this.y < 392) {
+            this.y += 83;
+    }
+
+};
+
+const defaultPlayerLocation = { x: 202, y: 392 };
+const player = new Player(defaultPlayerLocation, 'images/char-boy.png');
+
+const enemyLocations = [60, 143, 226]; // Y coordinates of each enemy position
```
Comment (OleksiyRudenko) [inline submissions/merowing/js-oop/app.js:75]: Mysterious numbers. Also this sort of comments means that naming could be better and making comment unnecessary.

File: submissions/merowing/js-oop/app.js [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+
+const randomNumber = (min = 200, max = 500) => {
+    return Math.round(Math.random() * (max - min) + min);
+};
+
+const allEnemies = enemyLocations.map(location => {
+    let speed = randomNumber();
+    return new Enemy(-101, location, speed);
```
Comment (OleksiyRudenko) [inline submissions/merowing/js-oop/app.js:83]: A mystery number.

**Review by OleksiyRudenko — CHANGES_REQUESTED** [review]
@merowing good job!
Although improvements are required.
Check the [requirements](https://kottans.org/documentation/docs/doc/code-review/#oo-js-frogger), in particular items 3 and 4.
Check your entire code against the requirements.

File: submissions/merowing/js-oop/app.js (hunk 1 above)
Comment (merowing) [reply submissions/merowing/js-oop/app.js]: Done.

File: submissions/merowing/js-oop/app.js (hunk 2 above)
Comment (merowing) [reply submissions/merowing/js-oop/app.js]: Fixed

File: submissions/merowing/js-oop/app.js (hunk 3 above)
Comment (merowing) [reply submissions/merowing/js-oop/app.js]: Fixed

File: submissions/merowing/js-oop/app.js [hunk 4]
```diff
@@ -0,0 +1,147 @@
...
+    WIDTH: 101,
+    SPRITE: "images/enemy-bug.png",
+  };
+  
+  const Enemy = function (x, y, speed) {
+    this.positionX = x;
+    this.positionY = y;
+    this.speed = speed;
+
+    this.player = player;
```
Comment (OleksiyRudenko) [inline submissions/merowing/js-oop/app.js:53]: A global variable.
The requirements provide a hint to resolution.

**Review by OleksiyRudenko — CHANGES_REQUESTED** [review]
@merowing almost there.
You moved "global variable" problem from one method to another method within the class.
But we need to eliminate the problem, not just push it around.

File: submissions/merowing/js-oop/app.js (hunk 4 above)
Comment (merowing) [reply submissions/merowing/js-oop/app.js]: done

**Review by OleksiyRudenko — APPROVED** [review]
@merowing great job!

**Pull request merged**